espresso/environment/2_espresso_liveness_test.go

```diff
@@ -119,3 +131,285 @@ func TestE2eDevNetWithEspressoEspressoDegradedLiveness(t *testing.T) {
 		}
 	}
 }
+
+// TestE2eDevNetWithEspressoEspressoDegradedLivenessViaCaffNode is a test that
+// checks that Espresso will return fast confirmations even when in a
+// degraded state.
+//
+// The criteria for this test is as follows:
+//	Requirement: Liveness:
+//    The rollup should continue to run, [to] post Espresso confirmations
+//    within 10 seconds of each rollup block produced by the sequencer.
+//
+// As a result, this test will submit a number of transactions to the sequencer,
+// while also consuming the Espresso stream of blocks utilizing the Espresso
+// streamer.  We **SHOULD** be able to match up the transactions submitted to
+// the blocks being produced by the Espresso Streamer, and the time it takes
+// from transaction submission to receiving the Block that contains that same
+// transaction should be less than 10 seconds.
+//
+// More importantly, this **SHOULD** also continue to be the state even when
+// Espresso is in a degraded state.
+//
+// Sadly, there does not seem to be an easy way to associate the Transaction
```

Comment:
Could this function help to achieve such a goal? https://github.com/EspressoSystems/optimism-espresso-integration/blob/1a2ec597438badb227320b0bcb58a69eba2c768c/op-e2e/e2eutils/geth/wait.go#L64

Just an idea no need to change you test that is working.

Reply:
Ah this initial comment was made under the idea that the data that was making up the `Batch` wasn't able to be correlated with the `Block`s coming from the `L2`.  I have been able to determine that I was incorrect in this understanding, and this comment should be adjusted to reflect that.

Reply:
Ah this initial comment was made under the idea that the data that was making up the Batch wasn't able to be correlated with the Blocks coming from the L2. I have been able to determine that I was incorrect in this understanding, and this comment should be adjusted to reflect that.

Reply:
Addressed in cdf5e56d23f7ae4b5ff688378e345e6d856e162f.